### DISCUSSION TITLE
Nextcloud Tracking Integration

### DISCUSSION THREAD

**Discussion by iamflorencejay** [plausible/analytics#1724] [Anything Else]
I love to see Nextcloud to track some stuff directly onto Plausible. Can you add this?

**Comment by metmarkosaric** [comment]
thanks for the feedback @iamflorencejay! what do you mean by this exactly?

**Comment by SemaphoreOxalis** [comment]
Personally, I can think something about the files opened (event have to be written here) because folders stats will not be very relevant because users have to go thought a lot of folders to reach the final one.

**Reply by metmarkosaric** [reply]
i see. does nextcloud allow you to insert javascript snippets into its pages?

**Reply by SemaphoreOxalis** [reply]
The easier I know is using this well named app : https://github.com/nextcloud/jsloader

**Reply by mjog** [reply]
FWIW I've tried a few times to get PA working with NC using that app and have never been successful.

**Comment by iamflorencejay** [comment]
Like for Piwik/Matomo, it has a nextcloud app that can track the visits, etc. I just want to have this feature on my arsenal.

**Reply by metmarkosaric** [reply]
you can try and use something like this to insert our snippet and see whether the data we provide is interesting enough for you: https://github.com/nextcloud/jsloader

we don't have any plans at the moment to build something custom for nextcloud tracking alone but our normal dashboard may be useful enough

**Reply by wohfab** [reply]
Did you have any luck with JSLoader, @iamflorencejay? When adding the Plausible snippet, I get this in the frontend and it doesn't send any data to Plausible:

```
<script defer src="[/apps/jsloader/js/script?v=9](https://DOMAIN/apps/jsloader/js/script?v=9)" nonce="...="></script>
```

**Reply by iamflorencejay** [reply]
> Did you have any luck with JSLoader, @iamflorencejay? When adding the Plausible snippet, I get this in the frontend and it doesn't send any data to Plausible:
> 
> ```
> <script defer src="[/apps/jsloader/js/script?v=9](https://DOMAIN/apps/jsloader/js/script?v=9)" nonce="...="></script>
> ```

didn't try that, but will try

**Reply by Normanras** [reply]
For anyone coming back to this, the JSLoader app needs to be updated. You can't install it on any Nextcloud instances past version 22.

**Reply by spacecake** [reply]
Pump, this is still an open issue.

**Comment by siment** [comment]
Hello from the future. For those of you using jsloader to inject Plausible tracking script, this is one way to do it. The trick is to dynamically inject a new script tag into HEAD from inside the function that jsloader wraps the javascript with. This is what you would paste into jsloader:

```javascript
const script = document.createElement('script');
var scriptUrl = 'https://plausible.example.com/js/script.js';
script.src = scriptUrl;
script.onload = () => {
    console.log(`${scriptUrl} loaded successfully!`);
    // You can call a callback function here if needed
};
script.onerror = () => {
    console.error(`Error loading ${scriptUrl}`);
};
document.head.appendChild(script);
```
